webapp/channels/src/components/channel_invite_modal/channel_invite_modal.tsx

```diff
@@ -79,161 +84,127 @@ export type Props = {
+    }, [props.membersInTeam, isGuest]);
+
+    // These functions are kept for potential future use but are not currently used in the component
+    // They were part of the original class implementation and might be needed for future features
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const removeInvitedUsers = useCallback((profiles: UserProfile[]) => {
+        setUsersNotInTeam((prevState) => {
+            const updatedUsers = prevState.filter((profile) => {
+                const user = profile as UserProfileValue;
+                const index = profiles.indexOf(user);
+                if (index === -1) {
+                    return true;
+                }
+                addValue(user);
+                return false;
+            });
+            return updatedUsers;
         });
-        this.setState({usersNotInTeam: [...usersNotInTeam], guestsNotInTeam: []});
-    };
-
-    public componentDidMount(): void {
-        this.props.actions.getProfilesNotInChannel(this.props.channel.team_id, this.props.channel.id, this.props.channel.group_constrained, 0).then(() => {
-            this.setUsersLoadingState(false);
+        setGuestsNotInTeam([]);
+    }, [addValue]);
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const removeUsersFromValuesNotInTeam = useCallback((profiles: UserProfile[]) => {
+        setUsersNotInTeam((prevState) => {
+            return prevState.filter((profile) => {
+                const index = profiles.indexOf(profile);
+                return index === -1;
+            });
         });
-        this.props.actions.getProfilesInChannel(this.props.channel.id, 0, USERS_PER_PAGE, '', {active: true});
-        this.props.actions.getTeamStats(this.props.channel.team_id);
-        this.props.actions.loadStatusesForProfilesList(this.props.profilesNotInCurrentChannel);
-        this.props.actions.loadStatusesForProfilesList(this.props.profilesInCurrentChannel);
-    }
-
-    public async componentDidUpdate(prevProps: Props, prevState: State) {
-        if (prevState.term !== this.state.term) {
-            const values = this.getOptions();
-            const userIds: string[] = [];
-
-            for (let index = 0; index < values.length; index++) {
-                const newValue = values[index];
-                if (this.isUser(newValue)) {
-                    userIds.push(newValue.id);
-                } else if (newValue.member_ids) {
-                    userIds.push(...newValue.member_ids);
-                }
-            }
-
-            if (!isEqual(values, this.state.groupAndUserOptions)) {
-                if (userIds.length > 0) {
-                    this.props.actions.getTeamMembersByIds(this.props.channel.team_id, userIds);
-                }
-                this.setState({groupAndUserOptions: values});
-            }
+        setGuestsNotInTeam([]);
+    }, []);
+
+    // Get excluded users
+    const getExcludedUsers = useCallback(() => {
```

Comment:
It would be better to use `useMemo` here instead of `useCallback` as the function only generates a set based on props and doesn't perform any operation.

Reply:
great catch, will be replaced in my next commit.